Create test for airgap env

donnyyung opened the pull request:
## Description

We want to run one test in an airgap environment to ensure OSS telepresence doesn't require anything from the internet to be successful. 

## Checklist

<!--
  Please review the requirements for each checkbox, and check them
  off (change "[ ]" to "[x]") as you verify that they are complete.
-->

 - [ ] I made sure to update `./CHANGELOG.md`.
 - [ ] I made sure to either submit a docs PR, or tell Matt about the necessary documentation changes.
 - [ ] My change is adequately tested.
 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.

Commit:
Better logging to debug tests not working macos

Signed-off-by: Donny Yung <donaldyung@datawire.io>

Commit:
different attempt

Signed-off-by: Donny Yung <donaldyung@datawire.io>

Commit:
unset SYSTEMA vars for tests

Signed-off-by: Donny Yung <donaldyung@datawire.io>

thallgren commented on .circleci/config.yml:
```diff
@@ -254,7 +254,11 @@ jobs:
       - run:
-          command: make test
+          command: |
+            echo "127.0.0.1 beta-app.datawire.io" | sudo tee -a /etc/hosts
```
A comment explaining what's going on here would be helpful.

thallgren commented on .circleci/config.yml:
```diff
@@ -254,7 +254,11 @@ jobs:
-          command: make test
+          command: |
+            echo "127.0.0.1 beta-app.datawire.io" | sudo tee -a /etc/hosts
+            echo "127.0.0.1 app.getambassador.io" | sudo tee -a /etc/hosts
```
Perhaps add `metriton.datawire.io` to the list to ensure that broken reporting doesn't mess things up.

thallgren commented on build-aux/main.mk:
```diff
@@ -193,7 +193,7 @@ format: $(tools/golangci-lint) $(tools/protolint) ## (QA) Automatically fix lint
 .PHONY: check
 check: $(tools/ko) $(tools/helm) ## (QA) Run the test suite
-	TELEPRESENCE_MAX_LOGFILES=300 go test -timeout=18m ./...
+	TELEPRESENCE_MAX_LOGFILES=300 SYSTEMA_HOST=127.0.0.1 TELEPRESENCE_LOGIN_DOMAIN=127.0.0.1 go test -timeout=18m ./...
```
Please add a comment that explains why these are set to localhost and perhaps add `SCOUT_HOST` to the list to ensure that broken reporting doesn't mess things up.

thallgren replied:
Hmm, I see that `SCOUT_HOST` is only used by `scout.py` so what's needed instead is to ensure that `https://metriton.datawire.io/scout` is redirected to localhost.

Commit:
Address PR feedback

Signed-off-by: Donny Yung <donaldyung@datawire.io>

thallgren reviewed (approved):
Looks good now, and reporting definitely fails :-)
```
2021/08/19 02:12:46.1714 error   connector/background-metriton : report failed: Post "https://metriton.datawire.io/scout": dial tcp 127.0.0.1:443: connect: connection refused
scout report
github.com/telepresenceio/telepresence/v2/pkg/client.(*Scout).Report
	/home/circleci/project/pkg/client/scout.go:196
github.com/telepresenceio/telepresence/v2/pkg/client/connector.run.func8
	/home/circleci/project/pkg/client/connector/command.go:457
github.com/datawire/dlib/dgroup.(*Group).goWorkerCtx.func1
	/home/circleci/.go_workspace/pkg/mod/github.com/datawire/dlib@v1.2.4-0.20210629021142-e221f3b9c3b8/dgroup/group.go:385
github.com/datawire/dlib/derrgroup.(*Group).Go.func2
	/home/circleci/.go_workspace/pkg/mod/github.com/datawire/dlib@v1.2.4-0.20210629021142-e221f3b9c3b8/derrgroup/errgroup.go:132
runtime.goexit
	/tmp/go/src/runtime/asm_amd64.s:1371
```

Commit:
Make systemaHost customizable in helm chart

Signed-off-by: Donny Yung <donaldyung@datawire.io>

Commit:
Add systema vars to config.yml

Signed-off-by: Donny Yung <donaldyung@datawire.io>

Commit:
Replace SYSTEMA_* env vars with vals in config.yml

Since some of these vars needed to be set in the helm chart (via the
config.yml), I figured I'd make all the places they are used come from
the config.yml

I left one TODO just because I'm about to be on vacation and don't want
to make a drastic change and then leave,so I'll get back to it once I'm
back.

Signed-off-by: Donny Yung <donaldyung@datawire.io>

Commit:
Update tests for airgap

Now that the SYSTEMA_* values come from the config.yml, we set them to
be localhost during the tests to ensure OSS Telepresence doesn't depend
on any extensions to function.

Signed-off-by: Donny Yung <donaldyung@datawire.io>
